File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        # return the max score of unique elems
        total = 0
        i = 0
        j = 0
        lookup = set()
        output = 0

        while (j < len(nums)):
            while (nums[j] in lookup):
                total -= nums[i]
                lookup.remove(nums[i])
                i += 1

            total += nums[j]
            lookup.add(nums[j])
            output = max(output, total)
            j += 1

        return output
        
        

File: 1813-maximum-erasure-value/maximum-erasure-value.js
var maximumUniqueSubarray = function(nums) {
    let lookup = new Set()
    let i = 0
    let j = 0
    let output = 0
    let currentSum = 0

    while (j < nums.length) {
        let num = nums[j]
        currentSum += nums[j]

        while (lookup.has(nums[j])) {
            lookup.delete(nums[i])
            currentSum -= nums[i]
            i += 1
        }

        lookup.add(num)
        output = Math.max(output, currentSum)
        j += 1
    }

    return output
};